### ISSUE TITLE
symbol's value as variable is void" error in markdown and latex

### ISSUE THREAD

**Issue by bdarcus** [jkitchin/org-ref#284]
I [reported this at spacemacs](https://github.com/syl20bnr/spacemacs/issues/7040), and they referred me here.

> The function `org-ref-helm-insert-cite-link` is autoloadable but the file in which it is defined does not load all the files it needs.

My original description:

If I open a markdown file and try to insert a citation, i get a "symbol's value as variable is void" error.

If I then open up an org file in the org layer, and do the same thing, I get expected behavior; helm-bibtex open up, etc.

If I then go back to the markdown layer, THEN it works correctly.

Same behavior with LaTeX.

In short, then, bibtex layer only works correctly if org-mode is first loaded.

**Comment by bdarcus** [comment]
And here's the [workaround](https://github.com/syl20bnr/spacemacs/pull/7047).

**Comment by TheBB** [comment]
To be precise, the file org-ref-helm-bibtex.el uses `org-ref-insert-cite-key`, defined in org-ref-core.el, which is not required by org-ref-helm-bibtex.el. I don't know if the intended usage is to manually require org-ref-core before calling or not, but the autoload cookie on the function creates an implicit promise that it is callable without further setup.

**Comment by jkitchin** [comment]
hm. Usually you should load org-ref with:

(require 'org-ref) which should load org-ref-core and load the backend you have defined.

I added a (require 'org-ref-helm) to org-ref-helm-bibtex.el in fc7ca7d9 that should fix this (that require should load org-ref-core) I think.

**Comment by bdarcus** [comment]
Great; thanks!
